src/flint/fmpz_mod_mat.jl

```diff
@@ -916,3 +916,38 @@ function matrix_space(R::ZZModRing, r::Int, c::Int; cached::Bool = true)
   ZZModMatrixSpace(R, r, c)
 end
 
+################################################################################
+#
+#  Kernel
+#
+################################################################################
+
+function AbstractAlgebra.Solve.kernel(M::ZZModMatrix; side::Symbol = :right)
+   AbstractAlgebra.Solve.check_option(side, [:right, :left], "side")
+
+   if side === :left
+      K = AbstractAlgebra.Solve.kernel(M, side = :right)
```

Comment:
In `HeckeMiscMatrix.jl` there is no `left_kernel` for this type, I just added the obvious. I assume this is correct, but wanted to ask because I never thought about linear algebra over rings with zero divisors (and never used a Howell form before).

Reply:
And by "obvious" I mean that, obviously, one should transpose the matrix. :man_facepalming: See the newest version.